packages/frontend/src/hooks/dashboard/useDashboardFiltersForExplore.ts

```diff
@@ -6,21 +6,23 @@ import {
 import { useCallback, useMemo } from 'react';
 import { useDashboardContext } from '../../providers/DashboardProvider';
 
-const useDashboardFiltersForExplore = (
-    tileUuid: string,
+function useDashboardFiltersForExplore(
+    tileUuid: string | undefined,
     explore: Explore | undefined,
-): DashboardFilters => {
-    const { dashboardFilters, dashboardTemporaryFilters } =
-        useDashboardContext();
+    failSilently: boolean = false,
```

Comment:
This seems like a red flag? What's the purpose of this? In what scenario do you expect this to fail?

Putting this here makes me think we have used the wrong abstraction somewhere upstream of this function

Reply:
@owlas good question, right now when we use a context that is not direct child of a context provider is throwing an error. we also have `/minimal/` pages for dashboards and charts that do not use those providers like `TrackingProvider` or `DashboardProvider`. that does not mean that chart should not render or throw an error.

I don't want to "prop drill" all the way down to the menu component and calling a react hook conditionally is a bad practice and causes bugs.

this seems like more explicit solution to it

so for `minimal` page this will just disable the menu rather than throw an error and it is not affecting any other places in the codebase

Reply:
open to suggestions if you have something better in mind that will be much safer alternative. Ideally this should be handled in the hook itself rather than outside of it. but we have a convention to make every hook "throw" an error if provider is not in the parent tree. that's why I decided to have an option to avoid throwing. maybe a naming change can improve this or another hook that does not throw an error?

Reply:
There's a couple of ways, the common pattern we've used is using a different context menu when on a dashboard: https://github.com/lightdash/lightdash/blob/main/packages/frontend/src/components/SimpleTable/index.tsx#L111-L120

However it could also be done by using component composition, providing context options to the component or optional dashboard options to the component, and these can be passed in further up the call stack

Reply:
updated! (created separate component to handle that)